### PR TITLE
[gradle] A flag to disable multimodule resources.

### PR DESCRIPTION
**Pull request by terrakok**
The Compose Multiplatform Resources with the mulimodule.publication support is a new feature and may have some not found problems. The PR introduces a gradle property to switch back to the previous behavior: `org.jetbrains.compose.resources.multimodule.disable=true`

## Testing
After update to Kotlin 2.0.0 and CMP 1.6.10 all projects will be switched to the new logic with the multimodule support.
Gradle info output should contain a message:
```
Configure KMP resources
```

If the new feature breaks user's projects they may add `org.jetbrains.compose.resources.multimodule.disable=true` to the `gradle.properties` to disable it. After that Gradle info output will contain:
```
Configure compose resources
``` 

### Resources
To disable the Compose Resources publication and the multimodule support in cases of problems add `org.jetbrains.compose.resources.multimodule.disable=true` to the root `gradle.properties`.

### CODE REVIEW

**Comment by igordmn** [comment]
@MatkovIvan, please, review as well

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ internal object ComposeProperties {
     internal const val MAC_NOTARIZATION_PASSWORD = "compose.desktop.mac.notarization.password"
     internal const val MAC_NOTARIZATION_TEAM_ID_PROVIDER = "compose.desktop.mac.notarization.teamID"
     internal const val CHECK_JDK_VENDOR = "compose.desktop.packaging.checkJdkVendor"
-    internal const val ALWAYS_GENERATE_RESOURCE_ACCESSORS = "compose.resources.always.generate.accessors"
+    internal const val ENABLE_MULTIMODULE_RESOURCES = "compose.resources.multimodule.enable"
```
Comment (igordmn) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: Is it about publication and depending on resource library, or only about publication?

**Comment by igordmn** [comment]
Release notes should in this format:

## Release Notes
### Breaking changes - Resources
- _(prerelease fix)_ To enable the Compose Resources publication and the multimodule support add `compose.resources.multimodule.enable=true` to the root `gradle.properties`.

File: gradle-plugins/compose/src/test/test-projects/misc/kmpResourcePublication/gradle.properties [hunk 2]
```diff
@@ -5,3 +5,4 @@ android.useAndroidX=true
 org.jetbrains.compose.experimental.uikit.enabled=true
 org.jetbrains.compose.experimental.jscanvas.enabled=true
 org.jetbrains.compose.experimental.wasm.enabled=true
+compose.resources.multimodule.enable=true
```
Comment (MatkovIvan) [inline gradle-plugins/compose/src/test/test-projects/misc/kmpResourcePublication/gradle.properties:8]: Shouldn't it be `org.jetbrains.*`?

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt [hunk 3]
```diff
@@ -22,7 +22,7 @@ internal object ComposeProperties {
     internal const val MAC_NOTARIZATION_PASSWORD = "compose.desktop.mac.notarization.password"
     internal const val MAC_NOTARIZATION_TEAM_ID_PROVIDER = "compose.desktop.mac.notarization.teamID"
     internal const val CHECK_JDK_VENDOR = "compose.desktop.packaging.checkJdkVendor"
-    internal const val ALWAYS_GENERATE_RESOURCE_ACCESSORS = "compose.resources.always.generate.accessors"
```
Comment (MatkovIvan) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: Why is it removed? It doesn't seem related

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt (hunk 1 above)
Comment (terrakok) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: It is about a publication and a multimoduleness (very close things in the gradle)

Comment (igordmn) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: What happen if it is `false`, and users add a dependency on a library with resources?

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt (hunk 3 above)
Comment (terrakok) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: It was forgotten unused constant

File: gradle-plugins/compose/src/test/test-projects/misc/kmpResourcePublication/gradle.properties (hunk 2 above)
Comment (terrakok) [reply gradle-plugins/compose/src/test/test-projects/misc/kmpResourcePublication/gradle.properties:8]: True. Missed that

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ComposeResources.kt [hunk 4]
```diff
@@ -41,23 +42,33 @@ private fun Project.onKgpApplied(config: Provider<ResourcesExtension>, kgp: Kotl
...
-            """
-                Compose resources publication requires Gradle >= $MIN_GRADLE_VERSION_FOR_KMP_RESOURCES
-                Current Gradle is ${currentGradleVersion.version}
-            """.trimIndent()
-        )
+        if (!enableMultimoduleResources) {
+            logger.info(
+                """
+                    Multimodule Compose Resources are disabled by default.
+                    To enable it add 'compose.resources.multimodule.enable=true' int the root gradle.properties file.
```
Comment (zsmb13) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ComposeResources.kt:55]: Small suggestion:
```suggestion
                    To enable it, add 'compose.resources.multimodule.enable=true' to the root gradle.properties file.
```

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt (hunk 1 above)
Comment (terrakok) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/ComposeProjectProperties.kt:25]: oh, you meant a library with resources (not the resource's library!). No, it requires the flag

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/KmpResources.kt [hunk 5]
```diff
@@ -20,7 +20,7 @@ internal fun Project.configureKmpResources(
     kotlinExtension as KotlinMultiplatformExtension
     kmpResources as KotlinTargetResourcesPublication
 
-    logger.info("Configure KMP resources")
+        logger.info("Configure KMP resources")
```
Comment (igordmn) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/KmpResources.kt:23]:
```suggestion
    logger.info("Configure KMP resources")
```

**Review by igordmn — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]
